src/account/AccountStorage.sol

```diff
@@ -79,6 +82,33 @@ function getPermittedCallKey(address addr, bytes4 selector) pure returns (bytes2
 
 using EnumerableSet for EnumerableSet.Bytes32Set;
 
+function toSetValue(FunctionReference functionReference) pure returns (bytes32) {
+    return bytes32(FunctionReference.unwrap(functionReference));
+}
+
+function toFunctionReference(bytes32 setValue) pure returns (FunctionReference) {
+    return FunctionReference.wrap(bytes21(setValue));
+}
+
+// HookData layout:
+// 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF______________________ Hook Function Reference
+// 0x__________________________________________AA____________________ is pre hook
```

Comment:
Perhaps we could precede the hook function with the type identifier or implement a sorting byte. This way, in storage, all prehooks could be organized before posthooks, allowing the loop to terminate early.

Reply:
I agree this would be useful in practice. Currently though, the spec doesn't mandate any ordering for hooks, so can we address this problem in a different PR, for this issue? https://github.com/erc6900/resources/issues/5

Reply:
👍  thanks!